candle-transformers/src/models/repvgg.rs

```diff
@@ -0,0 +1,233 @@
+//! RepVGG inference implementation
+//!
+//! See "RepVGG: Making VGG-style ConvNets Great Again" Ding et al. 2021
+//! https://arxiv.org/abs/2101.03697
+
+use std::collections::HashMap;
+
+use candle::{Result, Tensor, D};
+use candle_nn::{
+    batch_norm, conv2d_no_bias, linear, BatchNorm, Conv2d, Conv2dConfig, Func, VarBuilder,
+};
+
+#[derive(Clone)]
+struct RepVGGConfig {
+    a: f32,
+    b: f32,
+    groups: usize,
+    stages: [usize; 4],
+}
+
+#[rustfmt::skip]
+fn repvgg_config(cfg: &str) -> RepVGGConfig {
+    let configs = HashMap::from([
+        ( "A0", RepVGGConfig { a: 0.75, b: 2.5, groups: 1, stages: [2, 4, 14, 1]}),
+        ( "A1", RepVGGConfig { a: 1.0, b: 2.5, groups:1, stages: [2, 4, 14, 1]}),
+        ( "A2", RepVGGConfig { a: 1.5, b: 2.75, groups:1, stages: [2, 4, 14, 1]}),
+        ( "B0", RepVGGConfig { a: 1.0, b: 2.5, groups:1, stages: [4, 6, 16, 1]}),
+        ( "B1", RepVGGConfig { a: 2.0, b: 4.0, groups:1, stages: [4, 6, 16, 1]}),
+        ( "B2", RepVGGConfig { a: 2.5, b: 5.0, groups:1, stages: [4, 6, 16, 1]}),
+        ( "B3", RepVGGConfig { a: 3.0, b: 5.0, groups:1, stages: [4, 6, 16, 1]}),
+        ( "B1G4", RepVGGConfig { a: 2.0, b: 4.0, groups:4, stages: [4, 6, 16, 1]}),
+        ( "B2G4", RepVGGConfig { a: 2.5, b: 5.0, groups:4, stages: [4, 6, 16, 1]}),
+        ( "B3G4", RepVGGConfig { a: 3.0, b: 5.0, groups:4, stages: [4, 6, 16, 1]}),
+    ]);
+
+    configs[cfg].clone()
```

Comment:
Creating a hashmap and then extracting a single value out of it seems a bit odd, let's just do a normal pattern matching instead.

Reply:
It is weird indeed, unfortunately there can't be no global literal hashmaps which would have been a bit cleaner. It was the consequence of trying to parameterize the model via a string and not have to have a separate wrapper function for each of the 10 model variants.

Reply:
Replaced the hashmap with a match (on &str for now), it is indeed cleaner this way.